### PR TITLE
Add maven-wrapper and github actions / ci build

### PR DESCRIPTION
**Pull request by EugenMayer**
To ensure tests are running and proper PR support add gihub actions.

To have proper maven handling, add the wrapper.

Hint: the tests will fail due to the broken equals/hash implementation in `HealthServicesRequest` (since you added filter). Fixing that is the other PR #3

### CODE REVIEW

**Comment by EugenMayer** [comment]
@jon5477 maybe you can look at this one, i closed #3 since you fixed it yesterday. 

I think this one, adding gihub actions to run tests, would make sense nevertheless.

**Comment by jon5477** [comment]
Yes, I will be using GitHub Actions to build and publish the package onto GitHub Packages. I'll also need to update the documentation to include instructions on how to download the package.

**Review by jon5477 — COMMENTED** [review]
Can you rebase off the master branch and make adjustments from there?

File: pom.xml
Comment (jon5477) [inline pom.xml:1]: Is this change being made for a specific reason? Is it just to clarify what dependencies are test-related and not?

File: Makefile
Comment (jon5477) [inline Makefile:1]: Do we really need this file? Typically Makefile is reserved for C/C++ projects

File: .idea/vcs.xml
Comment (jon5477) [inline .idea/vcs.xml:1]: Please remove this file, I've updated the .gitignore to ignore all JetBrains IDE files

File: .idea/misc.xml
Comment (jon5477) [inline .idea/misc.xml:1]: Please remove this file, I've updated the .gitignore to ignore all JetBrains IDE files

File: .idea/encodings.xml
Comment (jon5477) [inline .idea/encodings.xml:1]: Please remove this file, I've updated the .gitignore to ignore all JetBrains IDE files

File: .github/workflows/build_test.yaml
Comment (jon5477) [inline .github/workflows/build_test.yaml:1]: Please rebase and merge your changes on the GH actions file I created, if you need this workflow to run on every push request I would like an explanation behind the reasoning on it (just curious)

**Pull request closed without merging**